### PR TITLE
take initial value prop

### PR DESCRIPTION
**Pull request by Jebpro**

### CODE REVIEW

File: src/QueryBuilder.jsx [hunk 1]
```diff
@@ -144,11 +146,11 @@ export default class QueryBuilder extends React.Component {
     }
 
     componentWillMount() {
-        const {fields, operators, combinators, controlElements, controlClassnames} = this.props;
+        const {fields, initValue, operators, combinators, controlElements, controlClassnames} = this.props;
```
Comment (pavanpodila) [inline src/QueryBuilder.jsx:149]: You are thinking right! I suggest we use the `this.props.query` itself to take in the initial value. There would be no need for a new prop.

File: src/QueryBuilder.jsx [hunk 2]
```diff
@@ -172,10 +174,11 @@ export default class QueryBuilder extends React.Component {
 
     }
 
-    getInitialQuery() {
-        return this.props.query || this.createRuleGroup();
+    getInitialQuery(initValue) {
+        return this.props.query || this.createRuleGroup(initValue);
```
Comment (pavanpodila) [inline src/QueryBuilder.jsx:178]: Here you can pass in `this.props.query` to `this.createRuleGroup()`.

**Review by pavanpodila — CHANGES_REQUESTED** [review]
Also need a unit test to prove this works ;-). Good job so far!

**Comment by Jebpro** [comment]
It seems that the query prop was exactly the missing piece I was looking for all along. It should be documented but when I tried to use the query prop on the component and I gave the query json to it, everything worked great. It populated the query builder with groups and rules just as I wanted. Therefore no PR for this feature is necessary as it does work but is just missing documentation.

**Comment by pavanpodila** [comment]
Can you make a PR to add that to readme? 😇
Many folks have asked for it!

Regards,

--
Pavan Podila

**Pull request closed without merging**